### DISCUSSION TITLE
cppcheck-vsg does not compile with Windows10 + visual studio 2019

### DISCUSSION THREAD

**Discussion by usr007** [vsg-dev/VulkanSceneGraph#1091] [General]
Hello, 
There is error message when compile project cppcheck-vsg with windows10 and visual studio 2019.
cppcheck: error: could not find or open any of the paths given, how did you resolve it?
It looks like the paths "include/vsg/*/*.h" and "vsg/*/*.cpp" passed to cppcheck is not expected.
It can be resolved by list all files and pass them to cppcheck within my test.
Regards.

**Comment by robertosfield** [comment]
I haven't ever tried cppcheck under VisualStudio.  This target is a utility I run regularly on Linux to help spot issues in the code base, it's not something I'd expect under end users to use.

Do you have cppcheck installed?  Is it something you want to use?

**Reply by usr007** [reply]
Yes, cppcheck is installed. 
       The error is not visual studio related, it is caused by the command as follow, which is generated by cmake.
G:\Tools\Cppcheck\cppcheck.exe -j 20 --quiet --enable=style --language=c++ --suppressions-list=F:/opensource/VulkanSceneGraph/cmake/cppcheck-suppression-list.txt F:/opensource/VulkanSceneGraph/include/vsg/*/*.h F:/opensource/VulkanSceneGraph/src/vsg/*/*.cpp -I F:/opensource/VulkanSceneGraph/include/
       All the paths are correct on my workstation, and it can be fixed by replace the two path parameters "F:/opensource/VulkanSceneGraph/include/vsg/*/*.h" and "F:/opensource/VulkanSceneGraph/src/vsg/*/*.cpp" with files listed from cmake file command.

No, It is not something I would actually use and matters. It comes from the first time I generate the solution with windows + visual studio 2019 and compile the solution.

**Comment by robertosfield** [comment]
I have created an issue #1092 for this problem.  

I have recently started using Windows for some development but hadn't seen this issue, I presume because I don't have cppcheck installed.  I presume the CMake macros for running cppcheck are doing something that's not quite portable.  As a workaround I could possible put the cppcheck target within the maintainer block of build targets.
